xikolo-ios/ViewControllers/Dashboard/CourseDatesTableViewController.swift

```diff
@@ -63,10 +72,25 @@ class CourseDatesTableViewController : UITableViewController {
 
     func updateAfterLoginStateChange() {
         self.navigationItem.rightBarButtonItem = UserProfileHelper.isLoggedIn() ? nil : self.loginButton
+        self.refresh()
+    }
 
+    func refresh() {
         self.tableView.reloadEmptyDataSet()
+        let deadline = 750.milliseconds.fromNow
```

Comment:
I guess this number is just randomly chosen?

Reply:
I found this time provides the a good UX (minimum time the refresh control is shown). Maybe this should be a constant.